### PR TITLE
test(aio): fix running docs examples against local builds

### PR DESCRIPTION
**Pull request by gkalpak**
This commit also updates the version of `@angular/cli` used for docs examples.

The previous (transient) dependency `@ngtools/webpack` was not compatible with `@angular/compiler-cli@>=5` and was breaking when running against the local builds (currently at 5.0.0-beta.2).

The version of `@ngtools/webpack` used by the latest `@angular/cli` version is compatible with `@angular/compiler-cli@5`.

### CODE REVIEW

**Comment by gkalpak** [comment]
BTW, once this is fixed, one can run the docs examples with the local builds with:

```
# Build the @angular packages
cd angular
npm install && ./builds.sh

# Install the examples' boilerplate (including dependencies)
cd aio
yarn && yarn boilerplate:add -- --local

# Build and serve an example locally
cd content/examples/<example-name>
npm start
```

File: aio/tools/examples/example-boilerplate.js [hunk 1]
```diff
@@ -41,12 +41,9 @@ const ANGULAR_PACKAGES = [
   'platform-browser-dynamic',
   'platform-server',
   'router',
+  'tsc-wrapped',
```
Comment (petebacondarwin) [inline aio/tools/examples/example-boilerplate.js]: Did someone move the tsc-wrapped folder?

File: aio/tools/examples/example-boilerplate.js [hunk 2]
```diff
@@ -111,7 +107,7 @@ class ExampleBoilerPlate {
     const sourceFolder = path.resolve(basePath, packageName);
     const destinationFolder = path.resolve(SHARED_NODE_MODULES_PATH, '@angular', packageName);
     shelljs.rm('-rf', destinationFolder);
-    fs.ensureSymlinkSync(sourceFolder, destinationFolder);
+    fs.copySync(sourceFolder, destinationFolder);
```
Comment (petebacondarwin) [inline aio/tools/examples/example-boilerplate.js]: Why this change?
Does it not work on Windows?
The idea of symlinking meant that you didn't need to keep running add-boilerplate all the time, while working on the source.

File: aio/tools/examples/run-example-e2e.js [hunk 3]
```diff
@@ -39,7 +39,7 @@ function runE2e() {
   if (argv.setup) {
     // Run setup.
     console.log('runE2e: copy boilerplate');
-    const spawnInfo = spawnExt('yarn', ['boilerplate:add', argv.local ? '-- --local': ''], { cwd: AIO_PATH });
+    const spawnInfo = spawnExt('yarn', ['boilerplate:add', '--', argv.local ? '--local' : ''], { cwd: AIO_PATH });
```
Comment (petebacondarwin) [inline aio/tools/examples/run-example-e2e.js]: Good spot.

File: aio/tools/examples/example-boilerplate.js (hunk 1 above)
Comment (petebacondarwin) [inline aio/tools/examples/example-boilerplate.js]: Ahah!
https://github.com/angular/angular/commit/619e625ee201d096f1c0801fd4d8d2d5d17c6ac0

File: aio/tools/examples/example-boilerplate.js (hunk 2 above)
Comment (gkalpak) [inline aio/tools/examples/example-boilerplate.js]: I can't try on Windows, but it didn't work on Ubuntu either. I was getting errors from files in `dist/packages-dist/...` that it couldn't find `@angular/core` for example. Seems like the double symlinking (packages-dist --> shared/node_modules --> <example>/node_modules) threw it off.

Comment (gkalpak) [inline aio/tools/examples/example-boilerplate.js]: Maybe we could get by, by copying the symlinks (instead of re-symlinking) from `shared/node_modules/@angular` if in `local` mode.

Comment (gkalpak) [inline aio/tools/examples/example-boilerplate.js]: Oh, that wouldn't actually work, because we symlink to the `shared/node_modules/` directory as a whole 😞

Comment (gkalpak) [inline aio/tools/examples/example-boilerplate.js]: So, I confirmed the symlinking [fails on Travis][1] too, so it was not a local issue in my Ubuntu VM.
Sticking with copying (and having to re-add the boilerplate if you re-build Angular packages).

[1]: https://travis-ci.org/angular/angular/jobs/260930986#L2876

**Comment by gkalpak** [comment]
Updated the `example-boilerplate.js` tests and ensured they are run on CI.

**Review by IgorMinar — APPROVED** [review]
